samples/client/petstore/java/resteasy-echo/src/test/java/org/openapitools/client/api/BodyApiTest.java

```diff
@@ -0,0 +1,210 @@
+
+import org.openapitools.client.ApiException;
+import java.io.File;
+import org.openapitools.client.model.Pet;
+import org.openapitools.client.model.Tag;
+import org.junit.Test;
+import org.junit.Ignore;
+import org.junit.Assert;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * API tests for BodyApi
+ */
+public class BodyApiTest {
+
+    private final BodyApi api = new BodyApi();
+
+    /**
+     * Test binary (gif) response body
+     *
+     * Test binary (gif) response body
+     *
+     * @throws ApiException
+     *          if the Api call fails
+     */
+    @Test
+    public void testBinaryGifTest() throws ApiException {
+        //
+        //File response = api.testBinaryGif();
+
+        // TODO: test validations
+    }
+    /**
+     * Test body parameter(s)
+     *
+     * Test body parameter(s)
+     *
+     * @throws ApiException
+     *          if the Api call fails
+     */
+    @Test
+    public void testBodyApplicationOctetstreamBinaryTest() throws ApiException {
+        //
+        //File body = null;
+        //
+        //String response = api.testBodyApplicationOctetstreamBinary(body);
+
+        // TODO: test validations
+    }
+    /**
+     * Test array of binary in multipart mime
+     *
+     * Test array of binary in multipart mime
+     *
+     * @throws ApiException
+     *          if the Api call fails
+     */
+    @Test
+    public void testBodyMultipartFormdataArrayOfBinaryTest() throws ApiException {
+        //
+        //List<File> files = null;
+        //
+        //String response = api.testBodyMultipartFormdataArrayOfBinary(files);
+
+        // TODO: test validations
+    }
+
+    /**
+     * Test single binary in multipart mime
+     *
+     * @see
+     * <a href="https://github.com/OpenAPITools/openapi-generator/issues/17367">#17367</a>
+     */
+    @Test
+    public void testBodyMultipartFormdataSingleBinaryTest()
+    throws ApiException, IOException {
+        File myFile = new File("test.txt");
+        if (!myFile.exists()){
+            assertTrue(myFile.createNewFile());
+        }
+        FileWriter fw = new FileWriter(myFile);
+        fw.write("testing only");
+        fw.close();
+        myFile.deleteOnExit();
+
+        String response =
+                api.testBodyMultipartFormdataSingleBinary(myFile);
+
+        EchoServerResponseParser parser =
+                new EchoServerResponseParser(response);
+
+        String contentDisposition = parser.headers.get("Content-Disposition");
+
+        assertThat(contentDisposition, containsString(
+                "form-data; name=\"my-file\"; filename=\"test.txt\""
+        ));
+    }
```

Comment:
thanks for the PR. I'l move this test to CustomTest.java instead in another PR (following the convention of other Java echo API client tests)

Reply:
great 👍 you’re welcome